cli/src/torii/routing.rs

```diff
@@ -262,7 +262,7 @@ async fn handle_post_configuration(
 }
 
 #[iroha_futures::telemetry_future]
-async fn handle_blocks_stream(sumeragi: Arc<Sumeragi>, mut stream: WebSocket) -> eyre::Result<()> {
+async fn handle_blocks_stream(kura: Arc<Kura>, mut stream: WebSocket) -> eyre::Result<()> {
```

Comment:
You're taking an `Arc` by value. Meaning that you atomically increment the reference counter and then decrement it. Accepting a `Weak` might require you to call an `expect` once, but it removes code from the hot path. In reality, I'd argue that given that there's always precisely one `Kura`, it could also be a global static with lazy initialisation, and interior mutability. What do you think @Erigara ? 

Also, regarding function signature, consider accepting an `impl Deref<Target = Kura>` instead. This will also allow you to switch it out for a `Weak<Kura>`, or a reference to a static.

Reply:
Ideally all our subsystems would be statics. But that breaks our tests because there would be just one peer per process.

Reply:
Maybe i miss something but i don't see how `Weak` would help, to access `Kura` we must call `upgrade` which will return `Arc<Kura>` and thus do increment/decrement.